test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc

```diff
@@ -444,6 +454,57 @@ TEST(QuicPlatformTest, QuicTestOutput) {
                       quic::QuicRecordTestOutput("quic_test_output.3", "output 3 content\n"));
 }
 
+TEST(QuicPlatformTest, QuicFlags) {
+  quiche::ResetFlags();
+  EXPECT_FALSE(GetQuicReloadableFlag(quic_testonly_default_false));
+  EXPECT_TRUE(GetQuicReloadableFlag(quic_testonly_default_true));
+  SetQuicReloadableFlag(quic_testonly_default_false, true);
+  EXPECT_TRUE(GetQuicReloadableFlag(quic_testonly_default_false));
+
+  EXPECT_FALSE(GetQuicRestartFlag(quic_testonly_default_false));
+  EXPECT_TRUE(GetQuicRestartFlag(quic_testonly_default_true));
+  SetQuicRestartFlag(quic_testonly_default_false, true);
+  EXPECT_TRUE(GetQuicRestartFlag(quic_testonly_default_false));
+
+  EXPECT_EQ(200, GetQuicFlag(quic_time_wait_list_seconds));
+  SetQuicFlag(quic_time_wait_list_seconds, 100);
+  EXPECT_EQ(100, GetQuicFlag(quic_time_wait_list_seconds));
+
+  quiche::ResetFlags();
+  EXPECT_FALSE(GetQuicReloadableFlag(quic_testonly_default_false));
+  EXPECT_TRUE(GetQuicRestartFlag(quic_testonly_default_true));
+  EXPECT_EQ(200, GetQuicFlag(quic_time_wait_list_seconds));
+  TCLAP::CmdLine cmdline("usage", '=');
+  quiche::RegisterFlags(cmdline);
+  std::vector<std::string> args{
+      "program name", "--quic_reloadable_flag_quic_testonly_default_false=1",
```

Comment:
If we do --quic_reloadable_flag_quic_testonly_default_false, without "=1", will it still be parsed as true?

Reply:
> > Meanwhile I do like the idea of folks being able to custom add command line flags, but this may be my in-google flag-loving bias - I'd like early design thoughts from a non-Googler (@mattklein123 , thoguhts?)
> 
> I agree custom adding CLI flags would potentially be nice, but I also agree from a quick read that most of these things seem like "runtime" settings? Should we instead be routing all of these settings via runtime?

I've moved to a mechanism that is not tied to command-line flags in the reboot of this PR, #6510. (That code is sufficiently different that it felt cleaner to draft as a new PR.) That PR provides an API for setting flags given the flag name and value as strings, available for runtime config to use in the future.